src/controllers/confirmation.controller.ts

```diff
@@ -0,0 +1,67 @@
+import { NextFunction, Request, Response } from "express";
+import {formatDateForDisplay} from "../client/date.formatter";
+import logger from "../logger";
+import {PTFCompanyProfile} from "../model/company.profile";
+import * as templatePaths from "../model/template.paths";
+import * as sessionService from "../services/session.service";
+import {ALREADY_SUBMITTED, COMPANY_NUMBER, COMPANY_PROFILE, SIGN_IN_INFO, USER_PROFILE} from "../session/keys";
+import {IUserProfile} from "../session/types";
+
+const ACCOUNTS_NEW_DEADLINE: number = 28;
+const CS_NEW_DEADLINE: number = 14;
+
+const createMissingError = (item: string): Error => {
+    const errMsg: string = item + " missing from session";
+    return new Error(errMsg);
+};
+
+const route = async (req: Request, res: Response, next: NextFunction): Promise<void> => {
+  const companyProfile: PTFCompanyProfile =
+      await sessionService.getPromiseToFileSessionValue(req.chSession, COMPANY_PROFILE);
+  const signInInfo = req.chSession.getSignedInInfo();
+  const userProfile: IUserProfile = signInInfo[USER_PROFILE] as IUserProfile;
+
+  const email = userProfile.email;
+  const dateRequested: Date = new Date(Date.now());
+  const extensionPeriodInDays =
+      (companyProfile.isAccountsOverdue) ? ACCOUNTS_NEW_DEADLINE : CS_NEW_DEADLINE;
```

Comment:
If neither accounts or confirmation statement are overdue, is this still the correct behaviour?

Reply:
This highlights an interesting point - presumably we are not going to get to this point if neither are overdue an alternative page should be displayed after clicking continue on the check company page stating this.

Reply:
Have added something to deal with this scenareo.